### DISCUSSION TITLE
SSL/TLS connection encryption

### DISCUSSION THREAD

**Discussion by RamanShuliak-Khazatech** [Havret/dotnet-activemq-artemis-client#501] [General]
Can anyone tell me if this form of setup is correct?

Here is broker acceptor settings:
`      <acceptor name="amqps-acceptor">tcp://0.0.0.0:61617?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=CORE,AMQP,STOMP,MQTT,OPENWIRE;useEpoll=true;sslEnabled=true;keyStoreAlias=amq1;keyStorePath=file:/certs/keystore.jks;keyStorePassword=11111111;enabledProtocols=TLSv1.2</acceptor>`

Here is logic of ConnectionFactory, Endpoint & Connection creation:
```
            connectionFactory = new ConnectionFactory()
            {
                SSL =
                {
                    ClientCertificates = null,
                    Protocols = SslProtocols.Tls12,
                    CheckCertificateRevocation = false,
                    RemoteCertificateValidationCallback = null,
                    LocalCertificateSelectionCallback = null
                }
            };

            var endpoint = Endpoint.Create(
                host: "localhost",
                port: 61617,
                user: "admin",
                password: "admin",
                scheme: Scheme.Amqps,
                path: "tcp://0.0.0.0:61617";

            endpoints.Add(endpoint);
            
            connection = await connectionFactory.CreateAsync(endpoints);
```
Thank you in advance)

**Comment by Havret** [comment]
It should be:

    var endpoint = Endpoint.Create(
                host: "localhost",
                port: 61617,
                user: "admin",
                password: "admin",
                scheme: Scheme.Amqps)
